File: packages/java/src/main/java/com/tesote/sdk/models/CursorPagination.java
package com.tesote.sdk.models;

import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
import com.fasterxml.jackson.annotation.JsonProperty;

/**
 * Pagination metadata for cursor-based endpoints (transactions list).
 *
 * <p>{@code afterId} is the last item in the page; {@code beforeId} is the
 * first. Pass {@code afterId} as {@code transactions_after_id} on the next
 * call to fetch the following page.
 */
@JsonIgnoreProperties(ignoreUnknown = true)
public record CursorPagination(
        @JsonProperty("has_more") Boolean hasMore,
        @JsonProperty("per_page") Integer perPage,
        @JsonProperty("after_id") String afterId,
        @JsonProperty("before_id") String beforeId
) {}

File: packages/php/src/Models/CursorPagination.php
<?php

declare(strict_types=1);

namespace Tesote\Sdk\Models;

/** Cursor pagination block (used by transactions index endpoints). */
final class CursorPagination
{
    public function __construct(
        public readonly bool $hasMore,
        public readonly int $perPage,
        public readonly ?string $afterId,
        public readonly ?string $beforeId,
    ) {
    }

    /**
     * @param array<string, mixed> $data
     */
    public static function fromArray(array $data): self
    {
        return new self(
            hasMore: (bool) ($data['has_more'] ?? false),
            perPage: (int) ($data['per_page'] ?? 0),
            afterId: isset($data['after_id']) ? (string) $data['after_id'] : null,
            beforeId: isset($data['before_id']) ? (string) $data['before_id'] : null,
        );
    }
}
